### PR TITLE
Add support for private key w/ passphrase to SSH client

### PR DESCRIPTION
**Pull request by ISauve**
### What does this PR do?

Currently the E2E SSH client only supports either 
(a) non-passphrase protected keys, or 
(b) keys stored in the unix SSH agent.
Which means windows hosts with passphrase-protected SSH keys are not currently supported - this PR fixes that.

Additionally, adds `privateKeyPath` and `privateKeyPassword` to the supported AWS config parameters in order to make configuration a little simpler: setting these values in `test_infra_config.yaml` would now look like
```
configParams:
  aws:
    publicKeyPath: <path to public key>
    privateKeyPath:  <path to private key>
    privateKeyPassword: <private key passphrase>
```
instead of
```
configParams:
  aws:
    publicKeyPath: <path to public key>
stackParams:
  ddinfra:
    aws/defaultPrivateKeyPath: <path to private key>
    aws/defaultPrivateKeyPassword: <private key passphrase>
```

### Additional Notes

An alternate solution could be to add support for the windows SSH agent ([like this)](https://github.com/DataDog/datadog-agent/pull/21257) - however, pulumi doesn't support the use of the windows SSH agent (see the [available connection configuration options here](https://github.com/pulumi/pulumi-command/blob/75ccc1443cbdc7e62bdcd82f9c0ee29f5b6e0923/sdk/go/command/remote/pulumiTypes.go#L18-L39)), which means that currently Windows users must already specify their private key & passphrase in order for pulumi to be able to establish the initial connection to test VMs. Re-using the same authentication mechanism in the SSH client here prevents the need for ssh keys to be configured in multiple places.

### Reviewer's Checklist

- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [x] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [x] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: test/new-e2e/pkg/runner/configmap.go [hunk 1]
```diff
@@ -107,6 +109,10 @@ func BuildStackParameters(profile Profile, scenarioConfig ConfigMap) (ConfigMap,
 	if err != nil {
 		return nil, err
 	}
+	err = SetConfigMapFromParameter(profile.ParamStore(), cm, parameters.PrivateKeyPassphrase, AWSPrivateKeyPassphrase)
```
Comment (vboulineau) [inline test/new-e2e/pkg/runner/configmap.go:112]:
```suggestion
	err = SetConfigMapFromParameter(profile.SecretStore(), cm, parameters.PrivateKeyPassphrase, AWSPrivateKeyPassphrase)
```

File: test/new-e2e/pkg/runner/parameters/const.go [hunk 2]
```diff
@@ -19,6 +19,8 @@ const (
 	ExtraResourcesTags StoreKey = "extra_resources_tags"
 	// KeyPairName config file parameter name
 	KeyPairName StoreKey = "key_pair_name"
+	// PrivateKeyPassphrase config file parameter name
+	PrivateKeyPassphrase StoreKey = "private_key_passphrase"
```
Comment (vboulineau) [inline test/new-e2e/pkg/runner/parameters/const.go:23]: To be consistent with existing test-infra naming.

```suggestion
	PrivateKeyPassword StoreKey = "private_key_password"
```

File: test/new-e2e/pkg/utils/e2e/client/vm_client.go [hunk 3]
```diff
@@ -48,10 +48,16 @@ func NewVMClient(t *testing.T, connection *utils.Connection, osType componentos.
 		}
 	}
 
+	privateKeyPassphrase, err := runner.GetProfile().ParamStore().GetWithDefault(parameters.PrivateKeyPassphrase, "")
```
Comment (vboulineau) [inline test/new-e2e/pkg/utils/e2e/client/vm_client.go:51]:
```suggestion
	privateKeyPassphrase, err := runner.GetProfile().SecretStore().GetWithDefault(parameters.PrivateKeyPassphrase, "")
```

**Review by vboulineau — APPROVED** [review]

File: test/new-e2e/pkg/runner/parameters/store_config_file.go [hunk 4]
```diff
@@ -103,6 +107,10 @@ func (s configFileValueStore) get(key StoreKey) (string, error) {
 		value = s.config.ConfigParams.AWS.KeyPairName
 	case PublicKeyPath:
 		value = s.config.ConfigParams.AWS.PublicKeyPath
+	case PrivateKeyPath:
+		value = s.config.ConfigParams.AWS.PrivateKeyPath
+	case PrivateKeyPassword:
```
Comment (pducolin) [inline test/new-e2e/pkg/runner/parameters/store_config_file.go:112]: **💬 suggestion**
You could add a test to `store_config_file.test`, you can update the fixture file at  `testfixtures/test_config_with_stackparams.yaml` and add an assert to `Test_NewConfigFileStore`

**Review by pducolin — APPROVED** [review]
Nice !

**Pull request merged**